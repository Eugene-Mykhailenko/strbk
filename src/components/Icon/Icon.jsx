import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import iconNamesFactory from './importIcons';

import styles from './Icon.scss';

export const iconNames = iconNamesFactory();

const Icon = (props) => {
  const { glyph, width, height, fill, className } = props;

  return (
    <svg
      className={classNames(className, styles.icon)}
      width={width}
      fill={fill}
      height={height}
    >
      <use xlinkHref={`#${glyph}`} />
    </svg>
  );
};

Icon.propTypes = {
  /** Icon name from Icons set. */
  glyph: PropTypes.string.isRequired,
  /** Icon width. */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Icon height. */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Default icon color. */
  fill: PropTypes.string,
  /** Additional class modifier for icon. */
  className: PropTypes.string,
};

Icon.defaultProps = {
  width: 14,
  height: 14,
  fill: 'currentColor',
  className: '',
};

Icon.styles = styles;

export default Icon;
