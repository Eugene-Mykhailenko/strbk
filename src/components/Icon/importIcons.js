export default function() {
  function importAllIcons(r) {
    r.keys().forEach(r);
  }

  const icons = require.context('../../assets/icons/', true, /.+\.svg$/);

  importAllIcons(icons);

  return icons
    .keys()
    .map((a) => a.replace('./', ''))
    .map((a) => a.replace('.svg', ''));
}
