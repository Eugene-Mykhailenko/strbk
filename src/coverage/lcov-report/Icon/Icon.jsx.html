<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Icon/Icon.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Icon</a> Icon.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import iconNamesFactory from './importIcons';
&nbsp;
import styles from './Icon.scss';
&nbsp;
export const iconNames = iconNamesFactory();
&nbsp;
const Icon = (props) =&gt; {
  const { glyph, width, height, fill, className } = props;
&nbsp;
  return (
    &lt;svg
      className={classNames(className, styles.icon)}
      width={width}
      fill={fill}
      height={height}
    &gt;
      &lt;use xlinkHref={`#${glyph}`} /&gt;
    &lt;/svg&gt;
  );
};
&nbsp;
Icon.propTypes = {
  /** Icon name from Icons set. */
  glyph: PropTypes.string.isRequired,
  /** Icon width. */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Icon height. */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Default icon color. */
  fill: PropTypes.string,
  /** Additional class modifier for icon. */
  className: PropTypes.string,
};
&nbsp;
Icon.defaultProps = {
  width: 14,
  height: 14,
  fill: 'currentColor',
  className: '',
};
&nbsp;
Icon.styles = styles;
&nbsp;
export default Icon;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 14 2018 16:36:00 GMT+0300 (EEST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
