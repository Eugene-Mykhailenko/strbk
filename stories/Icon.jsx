import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import storyStyles from './Stories.scss';

import Icon, { iconNames } from '../src/components/Icon/Icon';

const icons = iconNames.map((iconName) => (
  <Icon style={{ marginRight: 10 }} width="35" height="35" glyph={iconName} />
));

storiesOf('Icon', module)
  .add(
    'Icon',
    withInfo({
      text: `Icon component can be used for 
        rendering any SVG icon from predefined icons set.`,
      inline: true,
    })(() => (
      <div className={storyStyles.container}>
        <Icon width="25" height="25" glyph="UII_Bonuses" />
      </div>
    )),
  )
  .add('Icons set', () => (
    <div className={storyStyles.container}>
      <div className={storyStyles.col}>
        <h3 className={storyStyles.title}>All Icons</h3>
        <ul className={storyStyles['list-two-col']}>
          {icons.map((icon) => (
            <li className={storyStyles['list-item']} key={Math.random()}>
              {icon}
              {icon.props.glyph}
            </li>
          ))}
        </ul>
      </div>
    </div>
  ));
