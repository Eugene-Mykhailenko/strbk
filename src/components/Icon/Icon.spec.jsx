import React from 'react';
import Icon from './Icon';

jest.mock('./importIcons');

const config = {
  glyph: 'UII_Bonuses',
};

const xlinkHref = `#${config.glyph}`;

describe('Icon', () => {
  it('should be able to indicate xlinkHref attribute', () => {
    const icon = shallow(<Icon {...config} />);

    expect(icon.childAt(0).props().xlinkHref).toEqual(xlinkHref);
    expect(icon).toMatchSnapshot();
  });

  it('should be able to indicate fill attribute', () => {
    const icon = shallow(<Icon {...config} />);

    expect(icon.props().fill).toEqual('currentColor');
    expect(icon).toMatchSnapshot();
  });
});
