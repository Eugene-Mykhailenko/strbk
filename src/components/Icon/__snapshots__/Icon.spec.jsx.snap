// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should be able to indicate fill attribute 1`] = `
<svg
  className="icon"
  fill="currentColor"
  height={14}
  width={14}
>
  <use
    xlinkHref="#UII_Bonuses"
  />
</svg>
`;

exports[`Icon should be able to indicate xlinkHref attribute 1`] = `
<svg
  className="icon"
  fill="currentColor"
  height={14}
  width={14}
>
  <use
    xlinkHref="#UII_Bonuses"
  />
</svg>
`;
